File: app/src/main/java/com/stevenfrew/beatprompter/filter/AllSongsFilter.java
package com.stevenfrew.beatprompter.filter;

import com.stevenfrew.beatprompter.cache.SongFile;

import java.util.List;

public class AllSongsFilter extends SongFilter
{
    public AllSongsFilter(String name,List<SongFile> songs)
    {
        super(name,songs,true);
    }
    public boolean equals(Object o) {
        return o == null || o instanceof AllSongsFilter;
    }
}

File: app/src/main/java/com/stevenfrew/beatprompter/filter/AllSongsFilter.kt
package com.stevenfrew.beatprompter.filter

import com.stevenfrew.beatprompter.cache.SongFile

class AllSongsFilter(name: String, songs: MutableList<SongFile>) : SongFilter(name, songs, true) {
    override fun equals(other: Any?): Boolean {
        return other == null || other is AllSongsFilter
    }
}